app/assets/javascripts/common/utility.js

```diff
@@ -155,10 +155,131 @@ utility.validURL = function(str) {
 };
 
 
+utility.validFirstAndLastName = function(str){
+  // we allow suffixes like Jr. as well
+  return Boolean(str.match(/^[a-z,.'-]+\s[a-z,.'-]+(\s[a-z,.'-]+)?$/i));
+};
+
 /**
  * Determines if the browser has the ability to open and read files
  * @returns {Boolean} 
  */
-utility.fileOpeningAbilities = function() {
+utility.browserCanOpenFiles = function() {
     return (window.File && window.FileReader && window.FileList && window.Blob);
 };
+
+
+// STRING UTILITIES
+
+utility.capitalize = function(str){
+  // NOTE (@aguestuser):
+  // opted for util function isntead of polyfilling String.prototype
+  // b/c I did the latter and it produced a fatal namespace collision with datatables.js
+  return str.slice(0,1).toUpperCase() + str.slice(1);
+};
+
+// OBJECT UTILITIES
+
+utility.get = function(obj, key) {
+  var entry = utility.isObject(obj) && Object.getOwnPropertyDescriptor(obj, key);
+  return entry && entry.value;
+};
+
+utility.getIn = function(obj, keys){
+  return keys.reduce(
+    function(acc, key){ return utility.get(acc, key); },
+    obj
+  );
+};
+
+utility.set = function(obj, key, value){
+  var _obj = Object.assign({}, obj);
+  return Object.defineProperty(_obj, key, {
+    configurable: true,
+    enumerable: true,
+    writeable: true,
+    value: value
+  });
+};
+
+utility.setIn = function(obj, keys, value){
+  if (keys.length === 0) {
+    return value;
+  } else {
+    return utility.set(
+      obj,
+      keys[0],
+      utility.setIn(
+        utility.get(obj, keys[0]),
+        keys.slice(1),
+        value
+      )
+    );
+  }
+};
+
+utility.delete = function(obj, keyToDelete){
+  return Object.keys(obj).reduce(
+    function(acc, key){
+      return key === keyToDelete ?
+        acc :
+        utility.set(acc, key, utility.get(obj, key));
+    },
+    {}
+  );
+};
+
+utility.deleteIn = function(obj, keys){
+  var leafPath = keys.slice(0, -1);
+  var leafNode = utility.getIn(obj, leafPath);
+  return !(leafPath && leafNode) ?
+    obj :
+    utility.setIn(
+      obj,
+      leafPath,
+      utility.delete(leafNode, keys.slice(-1)[0])
+    );
+};
+
+// see https://github.com/paularmstrong/normalizr
+utility.normalize = function(arr){
+  return arr.reduce(
+    function(acc, item){ return utility.set(acc, item.id, item); },
+    {}
+  );
+};
+
+// mostly for use in deserializing JSON
+utility.stringifyValues = function(obj){
+  return Object.keys(obj).reduce(
+    function(acc, k){
+      var val = utility.get(obj, k);
+      return utility.set(
+        acc,
+        k,
+        // only stringify non-boolean existy vals
+        (val === true || val === false) ? val : val && String(val)
+      );
+    },
+    {}
+  );
+};
+
+utility.exists = function(obj){
```

Comment:
that *was* the intention. my reasoning was that `false` "exists" because it is a (boolean) value, whereas undefined or null do not exist. happy to reconsider.